src/service/video-manager-impl.js

```diff
@@ -868,12 +926,74 @@ class VideoEntry {
   }
 
   /**
+   * Get the autoplay state distinct from paused by user and paused by
+   * the video scrolling out of view.
+   * @return {boolean}
+   */
+  isAutoplay() {
+    return (this.getPlayingState() === PlayingStates.PLAYING_AUTO) ||
+        (this.playCalledByAutoplay_ && !this.userInteractedWithAutoPlay_);
+  }
+
+  /**
    * Returns whether the video was interacted with or not
    * @return {boolean}
    */
   userInteractedWithAutoPlay() {
     return this.userInteractedWithAutoPlay_;
   }
+
+  /**
+   * @param {string} eventType
+   * @param {!Object<string, string>=} opt_vars A map of vars and their values.
+   * @private
+   */
+  analyticsEvent_(eventType, opt_vars) {
+    const trackingVideo = assertTrackingVideo_(this.video);
+    if (trackingVideo) {
+      opt_vars = opt_vars || this.getAnalyticsDetails_(trackingVideo);
+      trackingVideo.element.dispatchCustomEvent(
+          VideoEvents.ANALYTICS, {type: eventType, details: opt_vars});
+    }
+  }
+
+  /**
+   * Collects a snapshot of the current video state for video analytics
+   * @param {!../video-interface.TrackingVideoInterface} video
+   * @return {!../video-interface.VideoAnalyticsDetailsDef}
+   * @private
+   */
+  getAnalyticsDetails_(video) {
+    const autoplay = this.isAutoplay();
```

Comment:
That will log whether the video was autoplay when it began playing and not when the analytics event occurred. I imagined that the publishers would want to know whether the event occurred while the video was autoplaying rather than how the video started. But if that's not the case I'm glad to change it.